### ISSUE TITLE
Unable to push any new changes after I sucessfully added custom attribute to cognito userpool

### ISSUE THREAD

**Issue by OukhouyaAyoub** [aws-amplify/amplify-cli#4980]
**Describe the bug**
I first added a custom attribute to my cognito userpool ( in the cloudformation template ) and it was pushed successfully and the custom attribute was implemented. 
After that I had to update a function ( custom message function trigger of the cognito userpool ) that I have also in my application, but when I push I get this error : 
`UPDATE_FAILED UserPool AWS::Cognito::UserPool Thu Jul 30 2020 19:45:57 GMT+0100 (Western European Summer Time) Existing schema attributes cannot be modified or deleted. (Service: AWSCognitoIdentityProvider; Status Code: 400; Error Code: InvalidParameterException; Request ID: null; Proxy: null)`
Even if I am not updating the userpool anymore and my previous update was pushed successfully in a different previous push

I also tried to push by category, amplify push function, but that also produces the same error. 
I also did an amplify env pull --restore and re applied my changes to the lambda function customMessage then pushed but still same error.
I am using amplify v 4.26.0
Any hints please ?

**Comment by OukhouyaAyoub** [comment]
I actually finally found the issue and therefore the solution.
So basically in the deployment S3 bucket `project_name-random_number-deployment` in the folder `amplify-cfn-templates/auth`  there was the wrong cloudformation template for the userpool with the old schema that wasn't updated (there is definitely an issue with the command amplify push). I uploaded to the bucket the correct template and it worked just fine!

**Comment by akshbhu** [comment]
Hi @OukhouyaAyoub 

Nice to know you resolved the problem. Can you tell the categories you added and changes you did when you faced this error?

**Comment by ammarkarachi** [comment]
Closing this issue, don't hesitate to reopen if you are still facing this issue.

**Comment by kldeb** [comment]
I came across this issue as well when trying to push changes to the auth category. It turned out that when we went through the `amplify auth update` questions the cli removed a custom field from the cf user pool template yaml file. Undoing that change allowed us to push the other updates. The cli should never remove a custom field once it's been deployed. 

I don't know if there's another ticket for this issue but there should be.
